### ISSUE TITLE
Port changes when am restarting dev server

### ISSUE THREAD

**Issue by aklinker1** [wxt-dev/wxt#1239]
> I've encountered this problem frequently since upgrade `"wxt": "^0.19.10"` to `"wxt": "^0.19.17"`, it always appears after modifying background script. 
> 
> It seems to be the problem of dev port, after reload script,  there is a log:  `[wxt] Connecting to dev server @ ws://localhost:3001`,  but it's still listening on 3000,  `3001` is not work, 
> 
> 
> My solution is assign the port number in  wxt config file: 
> ```ts
> export default defineConfig({
> 	dev:{
> 		server: {
> 			port:3000,
> 		}
> 
> 	},
> ...
> ```
> 
> _Originally posted by @slow-groovin in https://github.com/wxt-dev/wxt/issues/430#issuecomment-2510703827_

Introduced by #1176. Now that we started shutting down and restarting the server, it's trying to auto-discover the port, thinks 3000 is taken, so it chooses the next open one.

**Comment by gideonstele** [comment]
I encountered the same problem when I had upgrade `"wxt": "^0.19.13"` to `"wxt": "^0.19.17"`. I add the follow config to `wxt.config.ts` file to temporarily solve:

```typescript
defineConfig({
  // ...
  dev: {
    server: {
      port: 3000,
    },
  },
})
```

**Comment by aklinker1** [comment]
Meant to fix this yesterday, too busy though. This is a high priority

**Comment by aklinker1** [comment]
@gideonstele @slow-groovin Could you guys try out my fix? You can install the PR's build via:

```sh
npm i https://pkg.pr.new/wxt@1241
```

Make sure to remove your fixed port from the config.

I can't reproduce locally on my mac or linux machines. Might be a windows thing.

**Comment by gideonstele** [comment]
> @gideonstele @slow-groovin Could you try out my fix? You can install the PR's build via:
> 
> ```shell
> npm i https://pkg.pr.new/wxt@1241
> ```
> 
> Make sure to remove the fixed port from your config.
> 
> I can't reproduce locally on my mac or linux machines. Might be a windows thing.

My mechine is Macbook Pro M1Max with OS Sequoia 15.1.1 (24B91), env Node 20.18.0 pnpm 9.12.3

**Comment by aklinker1** [comment]
Ok, thanks for the info. Did you try out the fix?

**Comment by slow-groovin** [comment]
> @gideonstele @slow-groovin Could you try out my fix? You can install the PR's build via:
> 
> ```shell
> npm i https://pkg.pr.new/wxt@1241
> ```
> 
> Make sure to remove the fixed port from your config.
> 
> I can't reproduce locally on my mac or linux machines. Might be a windows thing.

I have tried,  firstly reproducing the problem with no port config,  then install this PR's build.  The problem is fixed.
My env is: Win10 10.0.19045,  Node 22.10.0,  Chromium 130.0.6723.31
My reproducing approach:  change a  import file  of background to make the extension hot-reload
```
i Changed: entrypoints\background\message.ts                                                                                                                                                                                
√ Reloaded extension      
```

**Comment by aklinker1** [comment]
Released in [`v0.19.19](https://github.com/wxt-dev/wxt/releases/tag/wxt-v0.19.19)

**Comment by gideonstele** [comment]
Sorry, I forgot to reply, the issue has been perfectly solved
